vfs/file.go

```diff
@@ -89,15 +96,21 @@ func (f *File) IsDir() bool {
 	return false
 }
 
+// IsSymlink returns true for symlinks when --links is enabled
+func (f *File) IsSymlink() bool {
+	return f.d.vfs.IsSymlink(f.Name())
+}
+
 // Mode bits of the file or directory - satisfies Node interface
 func (f *File) Mode() (mode os.FileMode) {
 	f.mu.RLock()
 	defer f.mu.RUnlock()
-	mode = f.d.vfs.Opt.FilePerms
-	if f.appendMode {
-		mode |= os.ModeAppend
+	// We could return the passed *create* FileMode, but it seems rclone want *always* Opt.FilePerms
```

Comment:
This doesn't look right - maybe the tests need fixing?

Reply:
No idea :|
It is hard for me to understand all correctly, there is lots of code in the VFS, so i have no idea how you imagined /designated the api initially, so that was more a question for you I added this comment :)

Reply:
Closing as part of the old trashed code